File: ws/src/main/scala/de/thm/ii/fbs/model/Course.scala
package de.thm.ii.fbs.model

/**
  * A course
  * @param name Name of the course
  * @param description The description of this course
  * @param visible The visibility of the course, false = invisible
  * @param id The id of the course, if 0, then none was assigned.
  */
case class Course(name: String, description: String = "", visible: Boolean = true, id: Int = 0)

File: modules/fbs-core/api/src/main/kotlin/de/thm/ii/fbs/model/v2/Course.kt
package de.thm.ii.fbs.model.v2

import javax.persistence.Column
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.OneToMany
import javax.persistence.Table

@Entity
@Table(name = "course")
class Course(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(nullable = false, name = "course_id", columnDefinition = "int")
    var id: Int,
    @Column(nullable = false, columnDefinition = "VARCHAR(100)")
    var name: String,
    @Column(nullable = false, columnDefinition = "TEXT")
    var description: String,
    @Column(nullable = false, columnDefinition = "TINYINT(1)")
    var visible: Boolean = true,
    @Column(columnDefinition = "int")
    val semesterId: Int? = null,
    @OneToMany(mappedBy = "course")
    val registration: Set<CourseRegisteration>
)
